sdk/src/azure/platform/az_posix.c

```diff
@@ -26,3 +30,129 @@ AZ_NODISCARD az_result az_platform_sleep_msec(int32_t milliseconds)
   (void)usleep((useconds_t)milliseconds * _az_TIME_MICROSECONDS_PER_MILLISECOND);
   return AZ_OK;
 }
+
+void az_platform_critical_error()
+{
+  while (1)
+  {
+  }
+}
+
+AZ_NODISCARD az_result az_platform_get_random(int32_t* out_random)
+{
+  _az_PRECONDITION_NOT_NULL(out_random);
+  
+  *out_random = (int32_t)random();
+  return AZ_OK;
+}
+
+void _thread_handler(union sigval sv)
```

Comment:
We should rename this to _timer_callback_handler maybe? It's not really a thread entry point but it does execute on the POSIX thread handling the timer event.

Reply:
I agree, renamed!